### ISSUE TITLE
Close ChangeStream Cursor in Reactive API

### ISSUE THREAD

**Issue by jaymell** [spring-projects/spring-data-mongodb#3773]
Howdy folks: 

I've recently discovered that if an error occurs in a Reactive change stream subscriber and the stream is retried, there seems to be no way to close prior change stream connections/cursors. As a result, my connection pool is occasionally getting filled with zombie connections. This is a contrived example but conveys the issue:

```
        template.getCollectionNames()
                .flatMap(collection -> template.changeStream(DB_NAME, collection, changeStreamOptions, Object.class))
                .flatMap(it -> Mono.error(new RuntimeException()))
                .retryWhen(Retry.fixedDelay(100000000, Duration.ofSeconds(1)))
```

A new connection is created every time the retry occurs, and eventually this will exhaust the connection pool, not to mention add a lot of additional read operations to the database. There seems to be no way to control this either using connection properties via `MongoClientSettings` or via the `MongoTemplate` or `MongoClient` APIs. 

Any assistance for the issue would be appreciated -- thank you.

**Comment by mp911de** [comment]
What MongoDB driver version are you using @jaymell?

**Comment by jaymell** [comment]
Hi, thanks for the reply. This issue occurs after testing the sample code with a retry delay of 5 seconds as well and is not intermittent, so I don't believe it's I/O-related.

We're currently using the 4.0.6 version of the MongoDB driver backed by a MongoDB 3.6 server.

**Comment by mp911de** [comment]
Thanks a lot. There has been an issue in the MongoDB issue tracker that describes the same scenario as reported here. It was solved with driver version 4.2.1 with https://jira.mongodb.org/browse/JAVA-3907. Please upgrade to a newer driver to eliminate the issue.
